### DISCUSSION TITLE
Is there a way to ssh with the `-t` argument?

### DISCUSSION THREAD

**Discussion by rawsp33d** [wezterm/wezterm#4709] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

sway

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

Is there a way to force a pseudo-terminal?

This would let me run `wezterm ssh` and cd into a specific directory, so if you have another way of doing this please feel free to suggest alternatives.

```bash
# works
ssh -t foo -- "cd ~/somedir; bash -l"

# no worky
wezterm ssh -t foo -- "cd ~/somedir; bash -l"
wezterm ssh foo -- "cd ~/somedir; bash -l"
```

I actually use nushell but I did try bash before asking this just in case that was the problem.

**Comment by nekowinston** [comment, marked as answer]
`wezterm ssh -o` with, I assume, `-oRequestTTY=force`;
```
Options:
  -o, --ssh-option <name=value>
          Override specific SSH configuration options. `wezterm ssh` is able to parse some (but not all!) options
          from your `~/.ssh/config` and `/etc/ssh/ssh_config` files. This command line switch allows you to
          override or otherwise specify ssh_config style options.
          
          For example:
          
          `wezterm ssh -oIdentityFile=/secret/id_ed25519 some-host`
```

**Comment by wez** [comment]
FWIW, `wezterm ssh` always requests a pty, so there is no need for a `-t` switch. In addition, the underlying `libssh` doesn't support the `RequestTTY` ssh config option; it simply does nothing.

However, I think the problem you're facing is a shell quoting issue.

```console
$ wezterm --config 'exit_behavior="Hold"' ssh foo -- "cd /tmp ; bash -l"
```

![image](https://github.com/wez/wezterm/assets/117777/58b84d42-0bdb-445f-8e7c-e09f9c9e20db)

but:

```console
$ wezterm --config 'exit_behavior="Hold"' ssh foo -- sh -c "cd /tmp; bash -l"
```

![image](https://github.com/wez/wezterm/assets/117777/f807148a-e928-4260-956e-822625cc591d)
